<?xml version="1.0"?>
<robot name="scara">
  
  <material name="blue">
    <color rgba="0 0 0.8 1"/>
  </material>
 

  <material name="orange">
    <color rgba="1 0.65 0 1"/>
  </material>
  
  
  <link name="link1">
    <visual>
      <geometry>
        <box size="0.25 0.25 0.4"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0.2"/>
      <material name="blue"/>    
    </visual>
  </link>
  
    
  <joint name="J1" type="revolute">
    <axis xyz="0 0 1"/>
    <limit effort="1000.0" lower="-2.5" upper="2.5" velocity="0.5"/>
    <origin rpy="0 0 0" xyz="0 0 0.5"/>
    <parent link="link1"/>
    <child link="link2"/>
  </joint>
  
  
    
  <link name="link2">
    <visual>
      <geometry>
        <box size="0.2 0.2 0.3"/>
      </geometry>
      <origin rpy="0 1.57079 0" xyz="0.125 0 0"/>
      <material name="orange"/>    
    </visual>
  </link>
  
  <joint name="J2" type="revolute">
    <axis xyz="0 0 1"/>
    <limit effort="1000.0" lower="-2" upper="2" velocity="0.5"/>
    <origin rpy="0 0 0" xyz="0.275 0 0"/>
    <parent link="link2"/>
    <child link="link3"/>
  </joint>
    
  
  <link name="link3">
    <visual>
      <geometry>
        <box size="0.15 0.15 0.3"/>
      </geometry>
      <origin rpy="0 1.57079 0" xyz="0.15 0 0"/>
      <material name="blue"/>    
    </visual>
  </link>
  
    
  <joint name="J3" type="prismatic">
    <axis xyz="0 0 1"/>
    <limit effort="1000.0" lower="0" upper="0.45" velocity="0.5"/>
    <origin rpy="0 0 0" xyz="0.325 0 0"/>
    <parent link="link3"/>
    <child link="jointHolder"/>
  </joint>
    
  
  <link name="jointHolder">
    <visual>
      <geometry>
        <cylinder length="0.01" radius="0.05"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="blue"/>   
    </visual>
  </link>  
  
  
  <joint name="J4" type="revolute">
    <axis xyz="0 0 1"/>
    <limit effort="1000.0" lower="-3" upper="3" velocity="0.5"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>   
    <parent link="jointHolder"/>
    <child link="link4"/>
  </joint> 
    

  <link name="link4">
    <visual>
      <geometry>
        <cylinder length="0.5" radius="0.05"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 -0.26"/>
      <material name="orange"/>   
    </visual>
  </link>  
  
  <joint name="left_gripper_joint" type="revolute">
    <axis xyz="0 0 1"/>
    <limit effort="1000.0" lower="0.0" upper="0.548" velocity="0.5"/>
    <origin rpy="0 1.57079 0" xyz="0 0 -0.5"/>
    <parent link="link4"/>
    <child link="left_gripper"/>
  </joint>

  <link name="left_gripper">
    <visual>
      <origin rpy="0.0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://urdf_tutorial/meshes/l_finger.dae"/>
      </geometry>
    </visual>
  </link>

  <joint name="left_tip_joint" type="fixed">
    <parent link="left_gripper"/>
    <child link="left_tip"/>
  </joint>

  <link name="left_tip">
    <visual>
      <origin rpy="0.0 0 0" xyz="0.09137 0.00495 0"/>
      <geometry>
        <mesh filename="package://urdf_tutorial/meshes/l_finger_tip.dae"/>
      </geometry>
    </visual>
  </link>
  
  <joint name="right_gripper_joint" type="revolute">
    <axis xyz="0 0 -1"/>
    <limit effort="1000.0" lower="0.0" upper="0.548" velocity="0.5"/>
    <origin rpy="0 1.57079 0" xyz="0 0 -0.5"/>
    <parent link="link4"/>
    <child link="right_gripper"/>
  </joint>

  <link name="right_gripper">
    <visual>
      <origin rpy="-3.1415 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://urdf_tutorial/meshes/l_finger.dae"/>
      </geometry>
    </visual>
  </link>

  <joint name="right_tip_joint" type="fixed">
    <parent link="right_gripper"/>
    <child link="right_tip"/>
  </joint>

  <link name="right_tip">
    <visual>
      <origin rpy="-3.1415 0 0" xyz="0.09137 0.00495 0"/>
      <geometry>
        <mesh filename="package://urdf_tutorial/meshes/l_finger_tip.dae"/>
      </geometry>
    </visual>
  </link> 
</robot>
